lib/grp.gi

```diff
@@ -4283,6 +4284,39 @@ local s,b;
   return s;
 end );
 
+#F  MakeGroupyType( <fam>, <filt>, <gens>, <id>, <isgroup> )
+# type creator function to incorporate basic deductions so immediate methods
+# are not needed. Parameters are family, filter to start with, generator
+# list, is it indeed a group (or only magma)?
+InstallGlobalFunction(MakeGroupyType,
+function(fam,filt,gens,id,isgroup)
+
+  if IsFinite(gens) then
+    if isgroup then
+      filt:=filt and IsFinitelyGeneratedGroup;
+    fi;
+
+    if Length(gens)>0 and CanEasilyCompareElements(gens) then
+      if id=false then
+        id:=One(gens[1]);
+      fi;
+      if id<>fail then # cannot do identity in magma
+        if ForAny(gens,x->x<>id) then
+          filt:=filt and HasIsTrivial and IsNonTrivial;
+          if isgroup and Length(gens)<=1 then # cyclic not for magmas
+            filt:=filt and IsCyclic;
+          fi;
+        else
+          filt:=filt and IsTrivial and HasIsNonTrivial;
```

Comment:
Since we now have implications IsTrivial => HasIsNonTrivial and IsNonTrivial => HasIsTrivial I'd kind of prefer if this line (and similar ones throughout this PR) were changed to e.g. `filt:=filt and IsTrivial` -- I find that much easier to grok; indeed, several times when reading through this PR did I misread a `HasIsTrivial` / `IsTrivial` / `HasIsNonTrivial` / `IsNonTrivial`